consensus/src/round_manager.rs

```diff
@@ -1096,6 +1096,11 @@ impl RoundManager {
                         }
                     };
                     proposals.sort_by_key(get_round);
+                    // If the first proposal is not for the next round, we only process the last proposal.
+                    // to avoid going through block retrieval of many garbage collected rounds.
+                    if self.round_state.current_round() + 1 < get_round(&proposals[0]) {
+                        proposals = vec![proposals.pop().unwrap()];
+                    }
```

Comment:
wasn't the point of going through all proposals to not require block retrieval/state sync? 

how does looking only at last proposal here help?